### ISSUE TITLE
How to get top levels of a project

### ISSUE THREAD

**Issue by RasmusGOlsen** [edaa-org/pyEDAA.ProjectModel#60]
The `TopLevel` attribute is declared in the following classes, `Project`, `Design`, and `FileSet`. How should the `project.TopLevel` be returned? Currently, it just returns `project.TopLevel` even if it's `None`. Shouldn't it search down the hierarchy for a value different than `None`?

**Comment by Paebbels** [comment]
Is there a typo in your questions?
> How should the project.TopLevel be returned? Currently, it just returns project.TopLevel even if it's None.

Or I didn't fully get the question.

At least `Design` and `FileSet` have indenpendent `TopLevel` attributes. The use case I have in mind for it is from Vivado. Here you have e.g. a VHDL design with a toplevel, but this includes IP cores. The IP core is encapsulated in a fileset and that IP core can be precompiled standalone as a netlist, which is then loaded by the main design. Such an IP core has its own local toplevel.

```
TopLevel.vhdl (global toplevel)
o- System.vhdl
   o- IPCore1.vhdl (local VHDL toplevel)
      o- Source1.vhdl
      o- Source2.vhdl
      o- Source3.vhdl
   o- IPCore2.v (local Verilog toplevel)
      o- Source1.v
      o- Source2.v
o- Clocking.vhdl
```

Right now, when writing these lines, I don't remember why a project has a toplevel, as there might be different design in a project with individual toplevels.

**Comment by RasmusGOlsen** [comment]
I agree that every `FileSet` can have a top-level. A top-level of a child `FileSet` should not propagate to its parent. But I think that for a top `FileSet` of a `Design`, the top levels should propagate to the `Design`. The `Design`s top level should propagate to the `Project`. I also think the `Project`s top-level type should be `List[str]` as multiple top levels should be possible. This is e.g. needed in simulation.

* Project [topA, topB, topC, topD]
  *  Design1  [topA, topB]
     * FileSetA TopA
        * FileSetA1 TopA1
     * FileSetB TopB
  * Design2 [topC, topD]
     * FileSetC TopC
     * FileSetD TopD

**Comment by Paebbels** [comment]
This sounds reasonable, I'll modify it.

Might need some time, because another Python library needs some fixes first. It's blocking releases on many other things.
